### PR TITLE
Add non power-of-two compressed texture support, and add mipmap support for compressed textures.

### PR DESCRIPTION
**Pull request by bythos14**
Also added some documentation and error checking. Switched RGB DXT1 to using recently added RGB1 BC1 from vitasdk, rather than RGBA BC1.

### CODE REVIEW

File: source/utils/gpu_utils.c [hunk 1]
```diff
@@ -314,7 +393,10 @@ void gpu_alloc_texture(uint32_t w, uint32_t h, SceGxmTextureFormat format, const
 			memset(texture_data, 0, tex_size);
 
 		// Initializing texture and validating it
-		sceGxmTextureInitLinear(&tex->gxm_tex, texture_data, format, w, h, 0);
+		if (sceGxmTextureInitLinear(&tex->gxm_tex, texture_data, format, w, h, 0) < 0) {
```
Comment (Rinnegatamante) [inline source/utils/gpu_utils.c:396]: Is the check here really necessary?

File: source/utils/gpu_utils.c [hunk 2]
```diff
@@ -533,6 +674,151 @@ void gpu_alloc_mipmaps(int level, texture *tex) {
...
+	uint32_t orig_h = sceGxmTextureGetHeight(&tex->gxm_tex);
+	SceGxmTextureFormat format = sceGxmTextureGetFormat(&tex->gxm_tex);
+	uint32_t bpp = 4;
+	uint32_t src_bpp;
+
+	uint32_t jumps[10];
+	w = nearest_po2(orig_w);
+	h = nearest_po2(orig_h);
+
+	if ((orig_w % 16 != 0) || (orig_h % 26 != 0)) {
```
Comment (Rinnegatamante) [inline source/utils/gpu_utils.c:694]: orig_h % 26 should be 16? Also this should go under a SKIP_ERROR_HANDLING i think.

File: source/utils/gpu_utils.c [hunk 3]
```diff
@@ -533,6 +674,151 @@ void gpu_alloc_mipmaps(int level, texture *tex) {
...
+		case GL_RGB:
+			read_cb = readRGB;
+			src_bpp = 3;
+			break;
+		default:
+			SET_GL_ERROR(GL_INVALID_OPERATION)
+			break;
+	}
+
+	if (temp[0] == NULL || temp[1] == NULL) {
```
Comment (Rinnegatamante) [inline source/utils/gpu_utils.c:750]: SKIP_ERROR_HANDLING

File: source/utils/gpu_utils.c [hunk 4]
```diff
@@ -533,6 +674,151 @@ void gpu_alloc_mipmaps(int level, texture *tex) {
...
+				src += src_bpp;
+				dst += bpp;
+			}
+		}
+	}
+
+	// Allocating the new texture data buffer
+	tex->mtype = use_vram ? VGL_MEM_VRAM : VGL_MEM_RAM;
+	void *texture_data = gpu_alloc_mapped(size, &tex->mtype);
+	if (texture_data == NULL) {
```
Comment (Rinnegatamante) [inline source/utils/gpu_utils.c:769]: SKIP_ERROR_HANDLING

**Review by Rinnegatamante — CHANGES_REQUESTED** [review]

**Comment by bythos14** [comment]
I'm not able to complete this PR currently, so I'll be closing it for now. If and when I get the chance, I'll try to return to it.

**Pull request closed without merging**